operator-wg/whitepaper-community-review/README.md

```diff
@@ -0,0 +1,875 @@
+
+### Multiple CRDs per controller
+Every CRD managed by an operator should be implemented in a single controller. This makes code a bit more readable and should help with separation of concerns.
+
+### Where to publish and find operators
+There are services like operatorhub.io and artifacthub.io which help end-users to find operators including instructions how they can be installed. This services often includes information about current security issues and the sources of operators. Additionally, information about the capabilities of operators is given.
+
+### Further reading
+There are lots of more best practices like:
+* An operator shouldn't install other operators
+* Operators shouldn't make assumptions about the namespaces they are deployed in, but also
+* Use an SDK for writing operators
+
+and many other best practices might be found on the internet. More of them could be found on following sources:
+* https://github.com/operator-framework/community-operators/blob/master/docs/best-practices.md
+* https://cloud.google.com/blog/products/containers-kubernetes/best-practices-for-building-kubernetes-operators-and-stateful-apps
+    
+## Designing Operators
+
+The previous chapter describes a use case for an operator that was one of the first operators ever. With no claim
+of completeness, this chapter deals with some best practice when writing
+own Operators, based on our own experience or described by the
+community. However, without clear knowledge of the actual state and
+without clear ideas of what we want to achieve, we also need some
+methods and techniques specifying what our Operator should do.
+Therefore, we will also have to deal with some aspects of requirement
+engineering.
+
+### Requirement analysis
+
+A key promise of Kubernetes is that it enables automation of operational
+tasks to deploy, scale, and manage containerized applications across
+multiple environments with no (or minimal) human intervention. In
+Kubernetes, stateless cloud-native applications are well suited for
+horizontal scaling, automated self-healing restarts, or progressive
+rollout of new containers. However, stateful applications with complex
+components running in clustered or distributed environments are not
+always well suited for this type of container-based infrastructure. They
+still require human interaction when it comes to persistence, upgrades,
+or high availability to remain in a stable state.
+
+True, Kubernetes solves these issues in a novel way by creating and
+managing custom applications using Operators. However, and here is the
+first question: as a developer, do you really know how this type of
+application works and interacts both internally and externally? How does
+the day-to-day IT operations work. How is the application backed up
+(including recovery). What steps are necessary in case of failovers or
+outages, are there any dependencies between the software components?
```

Comment:
```suggestion
True, Kubernetes solves these issues in a novel way by creating and
managing custom applications using Operators. However, and here is the
first question: as a developer, do you really know how this type of
application works and interacts both internally and externally? How do
the day-to-day IT operations work? How is the application backed up
(including recovery)? What steps are necessary in case of failovers or
outages, are there any dependencies between the software components?
```